samples/bluetooth/llpm/src/main.c

```diff
@@ -449,3 +455,29 @@ void main(void)
 		}
 	}
 }
+
+#if defined(CONFIG_BOARD_NRF52840DONGLE_NRF52840)
+static int usb_uart_init(const struct device *dev)
```

Comment:
It is not quite correct to initialize USB device stack this way. Better to move this function to the line 403 above `void main(void)`, and call it from the main function. There is no benefit use SYS_INIT at APPLICATION level. Also please remove `k_sleep(K_SECONDS(2));` because the application waits for DTR anyway. But sometimes it make sense to add k_sleep() to `while(!dtr)` loop to relax low priority thread situation, see [upstream's cdc_acm sample](https://github.com/zephyrproject-rtos/zephyr/blob/master/samples/subsys/usb/cdc_acm/src/main.c#L97) as example.

Reply:
yeah that's what I had before, but the device did not enumerate. I suppose some other thread is blocking on the console before it is initialized (i.e. before main executes)

Reply:
at least if what you meant was something like this:
```
void main(void)
{
	int err;
	static struct bt_conn_cb conn_callbacks = {
		.connected = connected,
		.disconnected = disconnected,
		.le_param_updated = le_param_updated,
	};

#if defined(CONFIG_BOARD_NRF52840DONGLE_NRF52840)
	const struct device *uart_dev = device_get_binding(
			CONFIG_UART_CONSOLE_ON_DEV_NAME);
	uint32_t dtr = 0;

	if (usb_enable(NULL)) {
		return -1;
	}

	/* Poll if the DTR flag was set, optional */
	while (!dtr) {
		uart_line_ctrl_get(uart_dev, UART_LINE_CTRL_DTR, &dtr);
		k_msleep(100);
	}
#endif

	printk("Starting Bluetooth LLPM example\n");

	console_init();

	bt_conn_cb_register(&conn_callbacks);

```

Reply:
The fact that this is not enumerating leads me to think that one of the options being enabled in the main [prj.conf](https://github.com/nrfconnect/sdk-nrf/blob/master/samples/bluetooth/llpm/prj.conf) file is blocking a thread before `main()`. I would disable those features one by one until we find the culprit, we should not merge the PR with the `SYS_INIT` workaround.

Reply:
It was CONFIG_LOG_MINIMAL that was the culprit. It should be investigated, in the meantime I'm just disabling it in the overlay.